server/types/base_ids.go

```diff
@@ -107,18 +107,30 @@ var baseIDArrayTypes = map[DoltgresTypeBaseID]DoltgresArrayType{}
 // baseIDCategories contains a map from all base IDs to their respective categories
 // TODO: add all of the types to each category
 var baseIDCategories = map[DoltgresTypeBaseID]TypeCategory{
-	Bool.BaseID():    TypeCategory_BooleanTypes,
-	BpChar.BaseID():  TypeCategory_StringTypes,
-	Float32.BaseID(): TypeCategory_NumericTypes,
-	Float64.BaseID(): TypeCategory_NumericTypes,
-	Int16.BaseID():   TypeCategory_NumericTypes,
-	Int32.BaseID():   TypeCategory_NumericTypes,
-	Int64.BaseID():   TypeCategory_NumericTypes,
-	Name.BaseID():    TypeCategory_StringTypes,
-	Numeric.BaseID(): TypeCategory_NumericTypes,
-	Oid.BaseID():     TypeCategory_NumericTypes,
-	Text.BaseID():    TypeCategory_StringTypes,
-	VarChar.BaseID(): TypeCategory_StringTypes,
+	AnyArray.BaseID():    TypeCategory_PseudoTypes,
+	Bool.BaseID():        TypeCategory_BooleanTypes,
+	Bytea.BaseID():       TypeCategory_UserDefinedTypes,
+	BpChar.BaseID():      TypeCategory_StringTypes,
+	Date.BaseID():        TypeCategory_DateTimeTypes,
+	Float32.BaseID():     TypeCategory_NumericTypes,
+	Float64.BaseID():     TypeCategory_NumericTypes,
+	Int16.BaseID():       TypeCategory_NumericTypes,
+	Int32.BaseID():       TypeCategory_NumericTypes,
+	Int64.BaseID():       TypeCategory_NumericTypes,
+	Json.BaseID():        TypeCategory_UserDefinedTypes,
+	JsonB.BaseID():       TypeCategory_UserDefinedTypes,
+	Name.BaseID():        TypeCategory_StringTypes,
+	Numeric.BaseID():     TypeCategory_NumericTypes,
+	Oid.BaseID():         TypeCategory_NumericTypes,
+	Text.BaseID():        TypeCategory_StringTypes,
+	Time.BaseID():        TypeCategory_DateTimeTypes,
+	Timestamp.BaseID():   TypeCategory_DateTimeTypes,
+	TimestampTZ.BaseID(): TypeCategory_DateTimeTypes,
+	TimeTZ.BaseID():      TypeCategory_DateTimeTypes,
+	Unknown.BaseID():     TypeCategory_UnknownTypes,
+	Uuid.BaseID():        TypeCategory_UserDefinedTypes,
+	VarChar.BaseID():     TypeCategory_StringTypes,
+	Xid.BaseID():         TypeCategory_UserDefinedTypes,
 }
```

Comment:
Similar to my other comment, this should also be something that is defined from within `DoltgresType`. The reason I didn't do it before was due to time constraints, and also due to the usage of `preferredTypeInCategory`. Since we're already making the changes for the base name, we may as well add these to the interface as well, and build them just like we're building `baseIDArrayTypes`. `preferredTypeInCategory` also needs to make sure that only one type is labeled as preferred, and panic if more than one type wants to be preferred.